server/config/machine-id.go

```diff
@@ -0,0 +1,23 @@
+package config
+
+import (
+	"crypto/md5"
+	"encoding/hex"
+	"os"
+
+	"github.com/denisbrodbeck/machineid"
+)
+
+func GetMachineID() string {
+	id, err := machineid.ProtectedID("tracetest")
+	// fallback to hostname based machine id in case of error
+	if err != nil {
```

Comment:
Small suggestion to avoid one more nesting: we can invert the `if` here and do an early return:

```golang
func GetMachineID() string {
    id, err := machineid.ProtectedID("tracetest")
    if err == nil {
      return id
    } 
    // do fallback case
}
```